oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/osh/OSHEntityTimeUtils.java

```diff
@@ -0,0 +1,357 @@
+package org.heigit.ohsome.oshdb.util.osh;
+
+import static java.lang.String.format;
+
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.heigit.ohsome.oshdb.osh.OSHEntity;
+import org.heigit.ohsome.oshdb.osh.OSHNode;
+import org.heigit.ohsome.oshdb.osh.OSHRelation;
+import org.heigit.ohsome.oshdb.osh.OSHWay;
+import org.heigit.ohsome.oshdb.osm.OSMEntity;
+import org.heigit.ohsome.oshdb.osm.OSMMember;
+import org.heigit.ohsome.oshdb.osm.OSMNode;
+import org.heigit.ohsome.oshdb.osm.OSMRelation;
+import org.heigit.ohsome.oshdb.osm.OSMWay;
+import org.heigit.ohsome.oshdb.util.OSHDBTimestamp;
+
+public class OSHEntityTimeUtils {
+  private OSHEntityTimeUtils() {
+    throw new IllegalStateException("utility class");
```

Comment:
any reasons to throw an exception in a private constructor? No one can call it any where accept the class it self.

Reply:
This is suggested by sonarlint ([rule S1118](https://sonarcloud.io/organizations/default/rules?languages=java&open=java%3AS1118&q=S1118)).  I believe the reasoning behind it might be that under some strange conditions this constructor might still be called (idk, maybe through _reflection_ or something similar).

Reply:
Ah, I see. The rule is about specify a private constructor, so that the default public is not generated, but the exception within is not necessary but also not harmful.